src/person/person.model.ts

```diff
@@ -206,6 +216,19 @@ PersonSchema.virtual('fullName').get(function () {
   return [this.firstName, this.lastName].filter(s => s).join(' ');
 });
 
+PersonSchema.pre('findOneAndUpdate', function () {
+  const hierarchy : string[] = (this as any).get('hierarchy');
+  const hierarchyPath : string[] = [];
+
+  if (hierarchy) {
+    hierarchy.forEach((elem: string, index: number) => {
+      hierarchyPath[index] = hierarchy.slice(index).join('/');
+    });
+  }
```

Comment:
when `hierarchy = [a,b,c]` the result will be `hierarchyPath=[a/b/c, b/c, c]` 
where it should be `hierarchyPath=[a, a/b, a/b/c]`